Fixed Tileset preview stretching textures

pyrophone opened the pull request:
Fix for #26526. Just replaces images smaller than (150, 100) by the width and height of the image for the rect. Larger images still use (150, 100).

![godot_ex](https://user-images.githubusercontent.com/15381918/53777054-f3ce4c00-3eef-11e9-8509-b45143b3ee4e.png)

pyrophone committed:
Fixed Tileset preview stretching textures

akien-mga commented on editor/plugins/tile_set_editor_plugin.cpp:
```diff
@@ -2918,7 +2918,12 @@ void TileSetEditor::update_texture_list_icon() {
 		RID rid = texture_list->get_item_metadata(current_idx);
 		texture_list->set_item_icon(current_idx, texture_map[rid]);
-		texture_list->set_item_icon_region(current_idx, Rect2(0, 0, 150, 100));
+		Size2 textureSize = texture_map[rid]->get_size();
```
We use `snake_case` convention for variable identifiers (as visible in the code around this), so it should be `texture_size`.

akien-mga commented on editor/plugins/tile_set_editor_plugin.cpp:
```diff
@@ -2918,7 +2918,12 @@ void TileSetEditor::update_texture_list_icon() {
-		texture_list->set_item_icon_region(current_idx, Rect2(0, 0, 150, 100));
+		Size2 textureSize = texture_map[rid]->get_size();
+
+		if (textureSize.x > 150 || textureSize.y > 100)
```
This will still stretch the texture if it's e.g. 50x200 (it will stretch to 150x100).

A better fix would be (removing the `if`):
```
texture_list->set_item_icon_region(current_idx, Rect2(0, 0, MIN(texture_size.x, 150), MIN(texture_size.y, 100)));
```

akien-mga commented:
For the above requested changes, please amend the commit with `git commit --amend` and force push the edited commit (`git push --force`) to update the PR (see [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html)).

Unrelated to these changes, but the email that you use to sign your commits (see https://patch-diff.githubusercontent.com/raw/godotengine/godot/pull/26622.patch) is not linked to your GitHub account, which is why it doesn't appear as authored by "@pyrophone". It's not a problem, but you can fix it by adding this email in your GitHub profile.

pyrophone commented:
I fixed the casing issue and switched to your fix instead of the if.

And thanks for pointing out that I didn't have my email displayed, I didn't realize it!

akien-mga commented on editor/plugins/tile_set_editor_plugin.cpp:
```diff
@@ -2918,7 +2918,8 @@ void TileSetEditor::update_texture_list_icon() {
 		texture_list->set_item_icon(current_idx, texture_map[rid]);
-		texture_list->set_item_icon_region(current_idx, Rect2(0, 0, 150, 100));
+		Size2 texture_size = texture_map[rid]->get_size();
+		texture_list->set_item_icon_region(current_idx, Rect2(0, 0, MIN(texture_size.x, 150), MIN(texture_size.y, 100)));;
```
There's a stray `;` at the end of the line.

pyrophone commented:
Just removed it and pushed.

akien-mga commented:
Thanks!
